src/debug-configuration/subproviders/pyocd-configuration-provider.ts

```diff
@@ -39,6 +41,23 @@ export class PyocdConfigurationProvider extends BaseConfigurationProvider {
         }
     }
 
+    protected resolveCmsisPackRootPath(target: TargetConfiguration): void {
+        const environmentValue = process.env.CMSIS_PACK_ROOT;
+        if (environmentValue) {
+            return;
+        }
+
+        if(target.environment?.CMSIS_PACK_ROOT) {
+            return;
+        }
+        const cmsisPackRootDefault = os.platform() === 'win32'
```

Comment:
Note for the future: we should start putting such desktop OS specifics into utility collections in the the `desktop` folder. Just in case we should expand to browser implementations one day. But let's worry about this later when we have more such helper functions which could give us guidance in how to organize them.